pkg/cvo/availableupdates.go

```diff
@@ -198,7 +204,17 @@ func (optr *Operator) SetArchitecture(architecture string) {
 	optr.architecture = architecture
 }
 
-func calculateAvailableUpdatesStatus(ctx context.Context, clusterID string, transport *http.Transport, upstream, arch, channel, version string) (configv1.Release, []configv1.Release, []configv1.ConditionalUpdate, configv1.ClusterOperatorStatusCondition) {
+func (optr *Operator) getDesiredArchitecture(update *configv1.Update) string {
+	if update != nil {
+		return string(update.Architecture)
+	}
+	return ""
```

Comment:
you could probably put the "default to current arch" fallback in here, although I don't think it matters much if you'd rather keep that defaulting up in the caller level.